bin/deku_node.re

```diff
@@ -132,19 +134,34 @@ let handle_block_level =
     Ok({level: Flows.find_block_level(Server.get_state())})
   });
 
+let empty_hash = Snapshots.{hash: BLAKE2B.hash(""), data: ""};
+let latest_snapshot = ref((false, empty_hash));
+
 let handle_protocol_snapshot =
   handle_request(
     (module Networking.Protocol_snapshot),
     (_update_state, ()) => {
       let State.{snapshots, _} = Server.get_state();
-      Ok({
-        snapshot: snd(snapshots.Snapshots.last_snapshot),
-        snapshot_hash: fst(snapshots.last_snapshot),
-        additional_blocks: snapshots.additional_blocks,
-        last_block: snapshots.last_block,
-        last_block_signatures:
-          Signatures.to_list(snapshots.last_block_signatures),
-      });
+      let.ok Snapshots.{hash: snapshot_hash, data: snapshot} =
+        switch (Snapshots.get_current_snapshot(snapshots)) {
+        | Some(snapshot) =>
+          latest_snapshot := (true, snapshot);
+          Ok(snapshot);
+        | None =>
+          let (initialized, latest_snapshot) = latest_snapshot^;
+          let.assert () = (`Node_not_yet_initialized, initialized);
+          Ok(latest_snapshot);
```

Comment:
This is a problem because of

https://github.com/marigold-dev/deku/blob/68170bf7315bb88493b86e6d7314a6047c495468/node/state.re#L191